eternalcore-core/src/main/java/com/eternalcode/core/feature/randomteleport/RandomTeleportSettingsImpl.java

```diff
@@ -0,0 +1,171 @@
+import net.dzikoysk.cdn.entity.Description;
+import org.bukkit.Material;
+
+@Contextual
+public class RandomTeleportSettingsImpl implements RandomTeleportSettings, DelaySettings {
+
+    @Description("# Time to wait for the random teleportation")
+    public Duration randomTeleportTime = Duration.ofSeconds(5);
+
+    @Description({
+        "# Type of random teleportation,",
+        "# WORLD_BORDER_RADIUS - radius based on the world-border size.",
+        "# STATIC_RADIUS - static radius based on the configuration.",
+    })
+    public RandomTeleportType randomTeleportType = RandomTeleportType.WORLD_BORDER_RADIUS;
+
+    @Description({
+        "# Radius of random teleportation, this uses for starting point spawn via /setworldspawn.",
+        "# If you want to use a static radius, set the type to STATIC_RADIUS and set the radius here.",
+        "# If you using WORLD_BORDER_RADIUS, this value will be ignored."
+    })
+    public RandomTeleportRadiusRepresenterImpl randomTeleportStaticRadius =
+        RandomTeleportRadiusRepresenterImpl.of(5000, 5000, 5000, 5000);
+    // For compatibility reasons, it must be named differently than "randomTeleportRadius".
+    // Due to limitations in the configuration library, changing the type of an existing field prevents the plugin from enabling.
+
+    @Description("# Teleport to a specific world, if left empty it will teleport to the player's current world")
+    public String randomTeleportWorld = "world";
+
+    @Description("# Number of attempts to teleport to a random location")
+    public int randomTeleportAttempts = 10;
+
+    @Description({
+        "# Unsafe blocks for random teleportation",
+        "# These blocks are considered unsafe for players to be teleported onto.",
+        "# The list includes blocks that can cause damage, suffocation, or other",
+        "# undesirable effects. Ensure that the list is comprehensive to avoid",
+        "# teleporting players to hazardous locations."
+    })
+    public Set<Material> randomTeleportUnsafeBlocks = EnumSet.of(
+        Material.LAVA,
+        Material.WATER,
+        Material.CACTUS,
+        Material.FIRE,
+        Material.COBWEB,
+        Material.SWEET_BERRY_BUSH,
+        Material.MAGMA_BLOCK,
+        Material.BEDROCK,
+        Material.TNT,
+        Material.SEAGRASS,
+        Material.TALL_SEAGRASS,
+        Material.BUBBLE_COLUMN,
+        Material.POWDER_SNOW,
+        Material.WITHER_ROSE
+    );
+
+    @Description({
+        "# Air blocks for random teleportation",
+        "# These blocks are considered safe for players to be teleported into.",
+        "# The list includes blocks that do not cause damage or impede movement.",
+        "# Ensure that the list is comprehensive to avoid teleporting players",
+        "# into solid or hazardous blocks."
+    })
+    public Set<Material> randomTeleportAirBlocks = EnumSet.of(
+        Material.AIR,
+        Material.CAVE_AIR,
+        Material.VOID_AIR,
```

Comment:
```suggestion
        Material.CAVE_AIR,
```

I'm not sure whether `CAVE_AIR` is supposed to be here, but I'm absolutely certain that `VOID_AIR` should not.

https://www.reddit.com/r/MinecraftCommands/comments/j3tby9/comment/g7e9ofk/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button

Reply:
"Void air‌ is used internally for blocks above (Y>319) and below (Y<-64) the world, and in unloaded chunks."

Source: https://minecraft.fandom.com/wiki/Air